src/lib/parser.c

```diff
@@ -2013,8 +2070,272 @@ RdbStatus elementModule(RdbParser *p) {
+                    consMeta.activeTime = consMeta.seenTime;
+                }
+
+                IF_NOT_OK_RETURN(rdbLoadLen(p, NULL, &(p->elmCtx.stream.consumer.pelLeft), NULL, NULL));
+
+                /*** ENTER SAFE STATE ***/
+
+                registerAppBulkForNextCb(p, bConsName);
+                CALL_HANDLERS_CB(p,NOP, RDB_LEVEL_DATA, rdbData.handleStreamNewConsumer, bConsName->ref, &consMeta);
+
+                p->elmCtx.stream.consumersLeft--;
+
+                DBG_RETURN(updateElementState(p, ST_LOAD_NEXT_CONSUMER_PEL, 0)); /* fall-thru */
+
+            case ST_LOAD_NEXT_CONSUMER_PEL:
+                while (p->elmCtx.stream.consumer.pelLeft) {
+                    BulkInfo *binfo;
+                    IF_NOT_OK_RETURN(rdbLoad(p, sizeof(RdbStreamID), RQ_ALLOC, NULL, &binfo));
+
+                    /*** ENTER SAFE STATE ***/
+
+                    streamDecodeID(binfo->ref, binfo->ref);
+
+                    CALL_HANDLERS_CB(p,NOP, RDB_LEVEL_DATA, rdbData.handleStreamConsumerPendingEntry, (RdbStreamID *) binfo->ref);
+                    p->elmCtx.stream.consumer.pelLeft--;
+                    DBG_RETURN(updateElementState(p, ST_LOAD_NEXT_CONSUMER_PEL, 0));
+                }
+                if (p->elmCtx.stream.consumersLeft)
+                    DBG_RETURN(updateElementState(p, ST_LOAD_NEXT_CONSUMER, 0));
+                else
+                    DBG_RETURN(updateElementState(p, ST_LOAD_NEXT_CONS_GROUP, 0));
+                break;
+
+            default:
+                RDB_reportError(p, RDB_ERR_STREAM_INVALID_STATE,
+                                "elementStreamLP() : Invalid parsing element state: %d.", ctx->state);
+                return RDB_STATUS_ERROR;
+                break;
+        }
+    }
+    return RDB_STATUS_OK;
+}
+
 /*** Loaders from RDB ***/
 
+/* This function loads a time from the RDB file. The reason for conditional
+ * conversion below is because before Redis 5 (RDB version 9), the function
+ * failed to convert data to/from little endian, so RDB files with keys having
+ * expires could not be shared between big endian and little endian systems
+ * (because the expire time will be totally wrong). The fix for this is just
+ * to call memrev64ifbe(), however if we fix this for all the RDB versions,
+ * this call will introduce an incompatibility for big endian systems:
+ * after upgrading to Redis version 5 they will no longer be able to load their
+ * own old RDB files. Because of that, we instead fix the function only for new
+ * RDB versions, and load older RDB versions as we used to do in the past,
+ * allowing big endian systems to load their own old RDB files. */
+RdbStatus rdbLoadMillisecTime(RdbParser *p, int64_t *val) {
+    BulkInfo *binfo;
+    IF_NOT_OK_RETURN(rdbLoad(p, 8, RQ_ALLOC, NULL, &binfo));
+    /*** ENTER SAFE STATE ***/
+    *val = *((int64_t *) binfo->ref);
+    if (p->rdbversion >= 9) /* see comment above */
+        memrev64ifbe(val);
+    return RDB_STATUS_OK;
+}
+
```

Comment:
Not suggesting a change here but endianness checks in redis reminds me this: https://commandcenter.blogspot.com/2012/04/byte-order-fallacy.html

In summary, instead of doing `memcpy/load + reverse`, if you do `byte shift + or`, you don't need to check endianness at all.

Reply:
I followed redis implementation. Better to be aligned at this stage. If you think there is an option to enhance redis code, then please consider review it file [rdb.c](https://github.com/redis/redis/blob/unstable/src/rdb.c).

Reply:
As I said, I'm not suggesting a change (too late for it). Just pointing another way for future reference.